### DISCUSSION TITLE
Is it intended to have only 1 user feedback in Sentry?

### DISCUSSION THREAD

**Discussion by Artein** [ueman/feedback#227] [General]
For some reason, I can have only 1 user feedback in Sentry. It actually shows multiple but under single user feedback. And it updates the title of the feedback to the most recent. This is super weird. So the most painful part is that if I resolve it, it resolves every feedback (cause it is 1 apparently). My blind guess is that every feedback sends with the same Id (but I am not sure).

Do you have any ideas on how to fix that?

**Comment by ueman** [comment]
This is because the Sentry integration is basically misusing the Sentry feedback functionality. Unfortunately, I don't think there's anything to improve it.

Though, you could try to play around with your grouping rules in Sentry.

**Reply by Artein** [reply]
Thank you for your prompt answer. I will think about what is possible to do in this case

**Reply by Artein** [reply]
Also would be nice to mention that "behavior" in README

**Comment by Artein** [comment]
I think the issue is that you don't provide a unique text to UserFeedback class. As far as I see they all NULL, thus Sentry groups them by that value, regardless of the description being provided in the feedback itself. As a fix you could generate and assign GUID there.

Here is the method in `feedback_sentry.dart`:
<img width="675" alt="SCR-20230503-lpee" src="https://user-images.githubusercontent.com/1109722/235886709-695be93c-7bfe-4fe3-834e-a0eeb0686dec.png">

This what I see in Sentry:
<img width="819" alt="SCR-20230503-lpmi" src="https://user-images.githubusercontent.com/1109722/235886735-31a26730-a5ca-4575-a02c-7dc1e97f12ad.png">

@ueman , what do you think?

**Reply by ueman** [reply]
Thanks for looking into it. Are you willing to fix it in a PR?

**Reply by Artein** [reply]
It depends on how quickly it will be merged into the master and released to pub.dev (this feature is not mandatory atm, but nice to have).

In any case, it is just an idea that should be tested first. That is why I wanted to validate it with you

**Reply by Artein** [reply]
Well, I think the issue is deeper than just hook-in before sending it to Sentry. Maybe something with simpleFeedbackBuilder...

As far as I got, BetterFeedback ctor receives FeedbackBuilder instance, and if none, passes simpleFeedbackBuilder. In my case I don't pass anything. So, will try to build my own FeedbackBuilder as a workaround for now.

EDIT: simpleFeedbackBuilder just returns instance of StringFeedback, so the issue most probably somewhere there

**Reply by Artein** [reply]
Disregard my previous messages. That NULL is for the UserFeedback.extras, so for now, I am out of ideas...

**Comment by Artein** [comment]
Solved. 

The thing is to add unique fingerprint to `captureMessage()` in `feedback_sentry.dart`:
```
OnFeedbackCallback sendToSentry({
  Hub? hub,
  String? name,
  String? email,
}) {
  final realHub = hub ?? HubAdapter();

  return (UserFeedback feedback) async {
    var uuid = Uuid();
    final id = await realHub.captureMessage(feedback.text, withScope: (scope) {
      scope.addAttachment(SentryAttachment.fromUint8List(
        feedback.screenshot,
        'screenshot.png',
        contentType: 'image/png',
      ));
      scope.fingerprint = ['feedback_${uuid.v1()}'];    <-- THIS LINE
    });
    await realHub.captureUserFeedback(SentryUserFeedback(
      eventId: id,
      email: email,
      name: name,
      comments: feedback.text + '\n${feedback.extra.toString()}',
    ));
  };
}
```

<img width="590" alt="SCR-20230503-npcd" src="https://user-images.githubusercontent.com/1109722/235914312-814aca6f-ce5a-4fad-9226-839f3d1045dd.png">

**Reply by Artein** [reply]
With this, your package will have an additional dependency — Uuid package

**Reply by ueman** [reply]
Sentry anyway already depends on that, so that's not a problem

**Comment by Artein** [comment]
@ueman, when will you be able to publish this small, yet crucial for this package, fix?

**Comment by KennyRakotoarimanana** [comment]
@ueman We also face this issue. Can you publish it?

**Comment by Artein** [comment]
@ueman ?

**Comment by ueman** [comment]
Sentry is currently working on new feedback functionalities. I don't plan to work on the current implementation in the feedback library until Sentry is done. After Sentry is done, I assume that feedback + Sentry will work much better together.

See https://github.com/getsentry/sentry-dart/pull/2230 and also https://github.com/getsentry/sentry-dart/issues/1593
